sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -1626,8 +1629,98 @@ def write_pipeline_spec_to_file(pipeline_spec: pipeline_spec_pb2.PipelineSpec,
+        'outputDefinitions': '# Outputs:'
+    }
+
+    def collect_pipeline_signatures(root_dict: dict,
+                                    signature_type: str) -> List[str]:
+        comment_strings = []
+        if signature_type in root_dict:
+            signature = root_dict[signature_type]
+            comment_strings.append(map_headings[signature_type])
+
+            # Collect data
+            array_of_signatures = []
+            for parameter in signature.get('parameters', {}):
+                data = {}
+                data['name'] = parameter
+                data['parameterType'] = map_parameter_types[
+                    signature['parameters'][parameter]['parameterType']]
+                if 'defaultValue' in signature['parameters'][parameter]:
+                    data['defaultValue'] = signature['parameters'][parameter][
+                        'defaultValue']
+                    if isinstance(data['defaultValue'], list) and not (
+                            isinstance(data['defaultValue'][0], dict) or
+                        (isinstance(data['defaultValue'][0], list))):
+                        data['defaultValue'] = sorted(data['defaultValue'])
+                    if isinstance(data['defaultValue'], dict) and not (
+                            isinstance(
+                                list(data['defaultValue'].keys())[0], dict) or
+                            isinstance(
+                                list(data['defaultValue'].keys())[0], list)):
+                        data['defaultValue'] = dict(
+                            sorted(data['defaultValue'].items()))
+                    if isinstance(data['defaultValue'], str):
+                        data['defaultValue'] = '"' + data['defaultValue'] + '"'
+                array_of_signatures.append(data)
+
+            for artifact in signature.get('artifacts', {}):
+                data = {}
+                data['name'] = artifact
+                data['parameterType'] = signature['artifacts'][artifact][
+                    'artifactType']['schemaTitle']
+                array_of_signatures.append(data)
+
+            array_of_signatures = sorted(
+                array_of_signatures, key=lambda d: d.get('name'))
+
+            # Present data
+            for signature in array_of_signatures:
+                string = '#    ' + signature['name'] + ': ' + signature[
+                    'parameterType']
+                if 'defaultValue' in signature:
+                    string += ' [Default: ' + str(
+                        signature['defaultValue']) + ']'
+                comment_strings.append(string)
+
+        return comment_strings
+
+    if 'root' not in pipeline_spec:
+        return ''
+
+    comment_sections = []
+    comment_sections.append('# PIPELINE DEFINITION')
+    comment_sections.append('# Name: ' + pipeline_spec['pipelineInfo']['name'])
+    if pipeline_description:
+        pipeline_description = '\n#              '.join(
```

Comment:
It looks like the length of this string is based on the size of the `'# Description'` string. Can you implement that programmatically to make the coupling explicit?